File: backend/src/main/java/com/project/entities/Professional.java
package com.project.entities;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(
    name = "professionals",
    uniqueConstraints = {
        @UniqueConstraint(columnNames = "user_id")
    }
)
public class Professional extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long professionalId;

    /**
     * User account linked to this professional
     */
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @NotNull
    private ProfessionalSpecialization specialization;

    @Enumerated(EnumType.STRING)
    @NotNull
    private SpokenLanguage spokenLanguage;

    @Min(0)
    private int experienceYears;

    @Lob
    private String qualification;

    @Lob
    private String bio;

    @DecimalMin("0.0")
    private BigDecimal consultationFee;

    private boolean isVerified = false;
}

File: backend_dotnet/SafeMind.Core/Entities/Professional.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using SafeMind.Core.Enums;

namespace SafeMind.Core.Entities
{
    [Table("professionals")]
    public class Professional : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("professional_id")]
        [JsonPropertyName("professionalId")]
        public long ProfessionalId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        [Column("user_id")]
        [JsonPropertyName("userId")]
        public long UserId { get; set; }

        [JsonPropertyName("user")]
        public User User { get; set; } = null!;

        [Required]
        [Column("specialization")]
        [JsonPropertyName("specialization")]
        public Specialization Specialization { get; set; }

        [Required]
        [Column("spoken_language")]
        [JsonPropertyName("spokenLanguage")]
        public SpokenLanguage SpokenLanguage { get; set; }

        [Range(0, 100)]
        [Column("experience_years")]
        [JsonPropertyName("experienceYears")]
        public int ExperienceYears { get; set; }

        [Column("qualification")]
        [JsonPropertyName("qualification")]
        public string? Qualification { get; set; } // @Lob -> string (max)

        [Column("bio")]
        [JsonPropertyName("bio")]
        public string? Bio { get; set; } // @Lob

        [Column("consultation_fee", TypeName = "decimal(18,2)")]
        [JsonPropertyName("consultationFee")]
        public decimal ConsultationFee { get; set; }

        [Column("is_verified")]
        [JsonPropertyName("isVerified")]
        public bool IsVerified { get; set; } = false;

        // public ICollection<ProfessionalAvailability> Availabilities { get; set; } = new List<ProfessionalAvailability>();
    }
}
